generator/ProtocolTestsGenerator/smithy-dotnet-codegen/src/main/java/software/amazon/smithy/dotnet/codegen/utils/ProtocolTestUtils.java

```diff
@@ -0,0 +1,34 @@
+package software.amazon.smithy.dotnet.codegen.utils;
+
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.dotnet.codegen.DotnetGenerationContext;
+import software.amazon.smithy.dotnet.codegen.DotnetSettings;
+
+/**
+ * Utility class for protocol tests
+ */
+public final class ProtocolTestUtils {
+    private ProtocolTestUtils(){}
+
+    /**
+     *
+     * @param  settings The settings object
+     * @see software.amazon.smithy.dotnet.codegen.DotnetSettings
+     * @return Returns the C# config class for a given test protocol.
+     */
+
+    public static Symbol getProtocolConfig(DotnetSettings settings){
+        //The packageNamespace for a protocol will always be in the format Amazon.<serviceName>
+        String namespace = settings.getPackageNamespace();
+        String[] packageNamespaceSplit = namespace.split("\\.");
+        if(packageNamespaceSplit.length != 2){
+            throw new CodegenException("The packageNamespace format in the smithy-build.json file is in the wrong format. Please make sure that is in the format" +
```

Comment:
I just put this check here for consistency. For example if the length was greater or less than 2, then we'd be using the wrong name for the Marshall Imports. The packageNamespace is used as the base namespace for the test services.

For example, if the packageNamespace is `Amazon.RestJsonProtocol`, we'd correctly import `Amazon.RestJsonProtocol.Model` etc..
but if the length > 2, then we might import something incorrectly like `Amazon.RestJsonProtocol.Foo.Model` which doesn't exist.

Reply:
I think this value would be derived from https://github.com/smithy-lang/smithy/blob/2479286123d79c1ab96981a68a6c7c43b0629f82/smithy-aws-protocol-tests/model/awsJson1_0/main.smithy#L11. Eventually when the full client generator is moved to Smithy instead of C2J, you'd need to use this to compute the packageNamespace. So I think you should remove it from Settings and add logic to compute this value from the model.

Reply:
Ah you're right that is a cleaner way to handle this, I'll update this